frontend/src/hooks/useZaakSelection.ts

```diff
@@ -131,43 +176,57 @@ export function useZaakSelection<T = unknown>(
   ) => {
     const optimisticSelection = zaken.reduce((selection, zaak) => {
       return { ...selection, [zaak.url as string]: { selected } };
-    }, zaakSelectionState);
-    setZaakSelectionState(optimisticSelection);
+    }, pageSpecificZaakSelection);
+    setPageSpecificZaakSelection(optimisticSelection);
   };
 
   /**
    * TODO: Optimize
    * @private
    */
-  const _updatePageSpecificZaakSelectionState = () => {
+  const _updatePageSpecificZaakSelectionState = async (
+    signal?: AbortSignal,
+  ) => {
+    if (!backend) return;
+
     // Get selection item for every zaak on page.
     const zaakUrls = zakenOnPage.map((z) => z.url as string);
-    const promises = zaakUrls.map((url) =>
-      getZaakSelectionItem(storageKey, url, false),
-    );
+    const newState = await getZaakSelectionItems<T>(
+      storageKey,
+      zaakUrls,
+      false,
+      backend,
+      { signal },
+    ).catch(_catchAbortError);
+
+    if (!newState) {
+      return;
+    }
+
+    // Update state.
+    const oldUrls = _serializeSelection(pageSpecificZaakSelection);
+    const newUrls = _serializeSelection(newState);
```

Comment:
I'm not sure, but possibly use the index of `zakenOnPage` to match to the results which may be off if the order changes, so I think it's the safe choice to update in such case. Possibly though it might be an optimization not to update if it's not needed, but this could make the check itself more expansive.

Unless you think otherwise I suggest leaving it as is.

Reply:
Yes, if for now the extra update is not a problem we can leave it